Applications/Spire/Source/Ui/EditableTableView.cpp

```diff
@@ -267,18 +267,22 @@ EditableTableView::EditableTableView(
 QWidget* EditableTableView::make_table_item(const ViewBuilder& view_builder,
     const std::shared_ptr<TableModel>& table, int row, int column) {
   if(column == 0) {
-    auto button = make_delete_icon_button();
-    button->setMaximumHeight(scale_height(26));
-    match(*button, DeleteButton());
-    auto tracker = std::make_shared<Tracker>(row);
-    tracker->m_connection = get_table()->connect_operation_signal(
-      std::bind_front(&TableRowIndexTracker::update, &tracker->m_index));
-    button->connect_click_signal([=] {
-      QTimer::singleShot(0, this, [=] {
-        delete_row(tracker->m_index);
+    if(row < table->get_row_size() - 1) {
```

Comment:
This looks awkward, help me understand it. The spec says that the last row should be a series of empty cells, if that's the case then that is the dominant condition to check. After that check, then you can check if the column is 0 and insert a delete button, and then after that check you can check if the column is the last column to insert an empty cell, and then finally as a last resort you use the view builder. If I misunderstand something please correct me.

Reply:
What component is contained in the empty cell of the last row?